server/apps/immich/src/api-v1/asset/asset.controller.ts

```diff
@@ -61,13 +66,24 @@ export class AssetController {
         const savedAsset = await this.assetService.createUserAsset(authUser, assetInfo, file.path, file.mimetype);
 
         if (uploadFiles.thumbnailData != null && savedAsset) {
-          await this.assetService.updateThumbnailInfo(savedAsset.id, uploadFiles.thumbnailData[0].path);
-          await this.backgroundTaskService.tagImage(uploadFiles.thumbnailData[0].path, savedAsset);
-          await this.backgroundTaskService.detectObject(uploadFiles.thumbnailData[0].path, savedAsset);
+          const assetWithThumbnail = await this.assetService.updateThumbnailInfo(
+            savedAsset,
+            uploadFiles.thumbnailData[0].path,
+          );
+
+          await this.assetUploadedQueue.add(
+            'asset-uploaded',
+            { asset: assetWithThumbnail, fileName: file.originalname, fileSize: file.size, hasThumbnail: true },
+            { jobId: savedAsset.id },
+          );
+        } else {
+          await this.assetUploadedQueue.add(
```

Comment:
This is nice but I think it applies to this case since the asset payload property of `asset` received two different payloads with different content. The content will be used later in the queue. If we try to keep the if statement simple, the refactor will be like this 

```typescript
        let assetWithThumbnail: AssetEntity;

        const savedAsset = await this.assetService.createUserAsset(authUser, assetInfo, file.path, file.mimetype);

        if (uploadFiles.thumbnailData != null) {
          assetWithThumbnail = await this.assetService.updateThumbnailInfo(
            savedAsset,
            uploadFiles.thumbnailData[0].path,
          );
        }

        await this.assetUploadedQueue.add(
          'asset-uploaded',
          {
            asset: uploadFiles.thumbnailData != null ? assetWithThumbnail : savedAsset,
            fileName: file.originalname,
            fileSize: file.size,
            hasThumbnail: uploadFiles.thumbnailData != null,
          },
          { jobId: savedAsset.id },
        );
```

There is less but also less verbose I think.

Reply:
Ah I see, yes I missed the different asset usage, original way is all good then, easier to understand what's going on that way